simplq/src/components/pages/Join/index.jsx

```diff
@@ -12,13 +12,18 @@ import { Banner } from '../Home/StaticInfos';
 import { JoinQButton } from '../../common/Button';
 import { handleEnterPress } from '../../common/utilFns';
 import InputField from '../../common/InputField';
+import PageNotFound from '../PageNotFound';
 
 export function JoinQueueWithDetails(props) {
   const queueId = props.match.params.queueId;
   const [queueStatusResponse, setQueueStatusResponse] = useState();
+  const [error, setError] = useState(false);
   useEffect(() => {
     async function fetchData() {
-      const response = await QueueService.getStatus(queueId).catch(handleApiErrors);
+      const response = await QueueService.getStatus(queueId).catch((e) => {
+        handleApiErrors(e);
+        setError(true);
+      });
       setQueueStatusResponse(response);
```

Comment:
Few things:

1) This would mean that whatever be the error, the page will redirect to 404. Can we redirect to 404 page only on token not found error from backend?

2) Related to (1), so if we move this logic into handleApiErrors, we can then centrally control places where we want to redirect to a 404 page.. what do you think? (404 page is, semantically thinking a response of handling a certain type of API error)

3) If we move the 404 page rendering logic into handleApiErrors, we can then avoid showing both the snac kbar and the 404 page together, only 404 page would do.


But let's make these changes after @thehamzarocks has done #164 ... deferring this change till then, let's move on... for now this is fine.

Reply:
For (1) I think it should redirect whatever the error because without the queue info response, the join queue page would make no sense. But like you said, we could think of another message to show instead of 404. Once the error handling is centralized, we can think of passing a heading dynamically to the current PageNotFound component instead of the static 404 heading.

Reply:
It should not redirect on all errors.. Errors that are not critical for loading the page need not be redirected. A good example of this is, when you enter a name and click on "Create Queue", the backend will return "The queue already exists, please use a different name". Here the snack bar should do, we don't need the redirect.

But if we are on the queue joining page, and there is no such queue, or same for token status page regarding tokens, then 404 page redirect

Reply:
This is just miscommunication. I was talking about this join queue situation in particular. For other cases, you're right, depends on the context.

Reply:
got it now.. then the simple algorithm I had in mind won't work. 

So, on certain API calls, it should go to 404 on any failure... so maybe what you are currently doing is the best way to go? For such calls, just `catch(e -> setError(true))` in the component where the call is made, and redirect to the 404 page. 

---

One thing that came to my mind... here you are setting a state variable "error" to true, and then ...

Instead can we just redirect to the 404 page on line 25?

Reply:
In our routes, we have defined the 404 component as a fallback/catch-all route:
```
<Route component={PageNotFound} />
```

so if we redirect to /pagenotfount or something like that... then that would work, if the catchall is working properly..

Reply:
Well the problem with redirecting to the 404 page is that the user can't single click the back button anymore to go back to the old page, since it would immediately redirect if the same error comes up. In this case with the join queue, that's tolerable since going back isn't useful, but we have to think if all cases are like that or not.

Reply:
I'll merge this anyway for now. We can check and merge the other PRs and do testing and think about the error handling parallelly.

Reply:
You are right... we can't redirect. 👍